ml-algorithms/src/main/java/org/opensearch/ml/engine/algorithms/remote/ConnectorUtils.java

```diff
@@ -245,4 +260,52 @@ public static SdkHttpFullRequest signRequest(
 
         return signer.sign(request, params);
     }
+
+    public static SdkHttpFullRequest buildSdkRequest(
+        Connector connector,
+        Map<String, String> parameters,
+        String payload,
+        SdkHttpMethod method
+    ) throws UnknownHostException {
+        String endpoint = connector.getPredictEndpoint(parameters);
+        Pattern pattern = Pattern.compile("(?:(\\w+)://)?((\\w+\\.)*\\w+)(?::(\\w+))?");
```

Comment:
I don't think it's necessary since this is only used in this method and appears once, only when it's been using for multiple places I think we need to move it to class static variable.